x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -294,21 +206,14 @@ export function ServiceOverviewTransactionsTable(props: Props) {
         <EuiFlexItem>
           <TableFetchWrapper status={status}>
             <ServiceOverviewTableContainer
-              isEmptyAndLoading={
-                items.length === 0 && status === FETCH_STATUS.LOADING
-              }
+              isEmptyAndLoading={transactionGroups.length === 0 && isLoading}
             >
               <EuiBasicTable
+                loading={isLoading}
```

Comment:
You may want to consider using `EuiInMemoryTable`. We also use it in `public/components/app/service_overview/service_overview_instances_table/index.tsx`. Then you don't have to slice the items yourself (but you still have to merge data from the second request, so it might not be worth it).

Reply:
I don't think it is worth changing to `EuiInMemoryTable`, since, as you said, I still need to slice the array to fetch its statistics. So I still would need to control the page information on the state.

Reply:
Hmm, I feel like we don't have to manage state for sort/pagination at the same level if we use an in-memory table, but I assume you've tried it out and it's not a blocker.